src/api/index.js

```diff
@@ -1,17 +1,32 @@
 import { Router } from 'express';
 import { version } from '../../package.json';
 
-// TODO: Implement mechanism to detect loggedIn users which yields profile data in /me endpoint.
+import User from '../db/models/user';
 
 export default () => {
   const api = Router();
 
+  function ensureAuthenticated(req, res, next) {
+    if (req.user && req.isAuthenticated()) return next();
+    res.sendStatus(401);
+  }
+
   api.get('/version', (req, res) => {
     res.json({ version });
   });
 
-  api.get('/me', (req, res) => {
-    res.json({ status: 'not implemented' });
+  api.get('/me', ensureAuthenticated, (req, res) => {
+    const query = User.findOne({ id: req.user.id }).exec();
+    query.then((queriedUser) => {
+      res.json({
+        name: queriedUser.name,
+        picture: queriedUser.picture,
+        updated: queriedUser.updated,
+        created: queriedUser.created,
+      })
+    }).catch((error) => {
+      res.json({ error: error });
```

Comment:
Consider adding error handling to a middleware rather than posting it here. Should be a 404 anyway.